birdhouse/pavics-compose.sh

```diff
@@ -157,44 +95,40 @@ done
 export AUTODEPLOY_EXTRA_REPOS_AS_DOCKER_VOLUMES
 
 # we apply all the templates
-find ./config ./components ./optional-components $EXTRA_CONF_DIRS -name '*.template' |
+find $ALL_CONF_DIRS -name '*.template' |
   while read FILE
   do
     DEST=${FILE%.template}
     cat ${FILE} | envsubst "$VARS" | envsubst "$OPTIONAL_VARS" > ${DEST}
   done
 
 if [ x"$1" = x"up" ]; then
-  # this is external in docker-compose.yml so have to create here
-  # no error if already exist, just an error message
-  docker network create jupyterhub_network
-
-  # no error if already exist
-  # create externally so nothing will delete these data volume automatically
-  docker volume create jupyterhub_data_persistence  # jupyterhub db and cookie secret
-  docker volume create thredds_persistence  # logs, cache
-
-  if [ ! -f "$GEOSERVER_DATA_DIR/global.xml" ]; then
-    echo "fix GeoServer data dir permission on first run only, when data dir do not exist yet."
-    FIRST_RUN_ONLY=1 $COMPOSE_DIR/deployment/fix-geoserver-data-dir-perm
-  fi
-
-  for adir in ${EXTRA_CONF_DIRS}; do
+  for adir in $ALL_CONF_DIRS; do
     COMPONENT_PRE_COMPOSE_UP="$adir/pre-docker-compose-up"
     if [ -x "$COMPONENT_PRE_COMPOSE_UP" ]; then
       echo "executing '$COMPONENT_PRE_COMPOSE_UP'"
       sh -x "$COMPONENT_PRE_COMPOSE_UP"
     fi
   done
-
 fi
 
 COMPOSE_CONF_LIST="-f docker-compose.yml"
-for adir in ${EXTRA_CONF_DIRS}; do
+for adir in $ALL_CONF_DIRS; do
   if [ -f "$adir/docker-compose-extra.yml" ]; then
     COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} -f $adir/docker-compose-extra.yml"
   fi
 done
+CONFIGURED_COMPONENTS=$(PROXY_SECURE_PORT=443 HOSTNAME=${PAVICS_FQDN} docker-compose ${COMPOSE_CONF_LIST} config --services)
+for adir in $ALL_CONF_DIRS; do
+  for extra_compose in "$adir"/docker-compose-extra-*.yml; do
+    service_name=$(basename "$extra_compose")
+    service_name=${service_name#docker-compose-extra-}
+    service_name=${service_name#.yml}
+    if echo "$CONFIGURED_COMPONENTS" | grep " $service_name "; then
+      COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} -f $extra_compose"
+    fi
```

Comment:
I'd like to propose using the same directory-based strategy as used for component configs instead of using derived components parts in file names of docker-compose extra fragments.

Assuming let's say that enabled components are [`magpie`, `twitcher`, `weaver`, `finch`].
Instead of looking for the following: 
``` 
birdhouse/components/finch/docker-compose-extra.yml
birdhouse/components/finch/docker-compose-extra-magpie.yml
birdhouse/components/weaver/docker-compose-extra.yml
birdhouse/components/weaver/docker-compose-extra-finch.yml
birdhouse/components/weaver/docker-compose-extra-magpie.yml
```

The script would look for:
<code><pre>birdhouse/components/_finch_/config/**finch**/docker-compose-extra.yml
birdhouse/components/_finch_/config/**magpie**/docker-compose-extra.yml
birdhouse/components/_weaver_/config/**weaver**/docker-compose-extra.yml
birdhouse/components/_weaver_/config/**finch**/docker-compose-extra.yml
birdhouse/components/_weaver_/config/**magpie**/docker-compose-extra.yml
</pre></code>

The reason why I want to make this proposition is because:
1. It is easier to know which part of the path corresponds to which component activation logic when parsing.
2. It is easier mentally to always think with directories, not having to switch between dir-based/named-based method since other files in configs, nginx, etc. are using this method.
3. Multiple components (eg: `weaver`, `cowbird`) already have nested `config/<dependency>` directories for other configurations they augment. Moving the `docker-compose-extra.yml` in those directories produces fewer changes.

Reply:
Sure that makes sense. I'll change that over today.

Reply:
done